### ISSUE TITLE
odbc mysql not working 

### ISSUE THREAD

**Issue by pratik1230** [processone/ejabberd#312]
hi 

i setup authentication odbc , when i register user from command line user registered but not inserting in mysql DB
and i tried with server localhost/127.0.0.1 but not working

i have installed this in amazone ec2 instance.

here is my .yml file

``` yaml
loglevel: 4
log_rotate_size: 10485760
log_rotate_date: ""
log_rotate_count: 1
log_rate_limit: 100
hosts:
  - "54.68.179.228"
listen: 
  - 
    port: 5222
    module: ejabberd_c2s
    max_stanza_size: 65536
    shaper: c2s_shaper
    access: c2s
    starttls: true
    certfile: "/opt/ejabberd-14.07/conf/server.pem"
  - 
    port: 5269
    module: ejabberd_s2s_in
    max_stanza_size: 131072
    shaper: s2s_shaper
  - 
    port: 5280
    module: ejabberd_http
    web_admin: true
    http_poll: false
    http_bind: true
    captcha: false
 auth_method: odbc
 odbc_type: mysql
 odbc_server: "127.0.0.1"
 odbc_database: "chat"
 odbc_username: "root"
 odbc_password: "1234"
shaper:
  normal: 1000
  fast: 50000
max_fsm_queue: 1000
acl:
  admin:
    user:
      - "admin": "54.68.179.228"
  local: 
    user_regexp: ""
access:
  max_user_sessions: 
    all: 10
  max_user_offline_messages: 
    admin: 5000
    all: 100
  local: 
    local: allow
  c2s: 
    blocked: deny
    all: allow
  c2s_shaper: 
    admin: none
    all: normal
  s2s_shaper: 
    all: fast
  announce: 
    admin: allow
  configure: 
    admin: allow
  muc_admin: 
    admin: allow
  muc_create: 
    local: allow
  muc: 
    all: allow
  pubsub_createnode: 
    local: allow
  register: 
    all: allow
  trusted_network: 
    loopback: allow
 registration_timeout: infinity
language: "en"
modules: 
  mod_adhoc: []
  mod_announce: # recommends mod_adhoc
    access: announce
  mod_caps: []
  mod_carboncopy: []
  mod_configure: [] # requires mod_adhoc
  mod_disco: []
  mod_http_bind: []
  mod_last: [{db_type: odbc}]
  mod_muc: 
    access: muc
    access_create: muc_create
    access_persistent: muc_create
    access_admin: muc_admin
   db_type: odbc
  mod_offline: 
    access_max_user_messages: max_user_offline_messages
     db_type: odbc
  mod_privacy: [{db_type: odbc}]
  mod_private: [{db_type: odbc}]
  mod_pubsub: 
    access_createnode: pubsub_createnode
db_type: odbc
    ignore_pep_from_offline: true
    last_item_cache: false
    plugins: 
      - "flat"
      - "hometree"
      - "pep" # pep requires mod_caps
  mod_register: 
    welcome_message: 
      subject: "Welcome!"
      body: |-
        Hi.
        Welcome to this XMPP server.
    ip_access: 
    access_form: register_form
  mod_roster: []
  mod_shared_roster: []
  mod_vcard: [{db_type: odbc}]
  mod_version: []
```

**Comment by saifromeo** [comment]
is this resolved, i am also having similar problem now

**Comment by mremond** [comment]
As @zinid said, it was incorrect syntax in configuration file, so yes, it is solved.

**Comment by saifromeo** [comment]
i am going through this problem, and breaking the head for an hr now
could not resolve, any help would be appreciated, my skype :saifromeo

**Comment by saifromeo** [comment]
this is what I did, I had an EC2 instance running at aws, worked fine on default mnesia db. today i wanted to experiment connecting mysql db. I spun an rds instance and imported the ejabberd sql into it. I followed the steps changing db parameters, few modules appened with odbc etc.., and then i am getting error 

2016-03-14 10:13:16.213 [error] <0.38.0>@ejabberd_config:get_plain_terms_file:222 Cannot load /opt/ejabberd-16.02/conf/ejabberd.yml: Syntax error on line 334 at position 4: did not find expected key

@ line 334, i have - odbc_type: mysql
